src/util/gas-factory-helpers.ts

```diff
@@ -543,3 +544,161 @@ export function initSwapRouteFromExisting(
     portionAmount: swapRoute.portionAmount,
   };
 }
+
+export const calculateL1GasFeesHelper = async (
+  route: RouteWithValidQuote[],
+  chainId: ChainId,
+  usdPool: Pair | Pool,
+  quoteToken: Token,
+  nativePool: Pair | Pool | null,
+  l2GasData?: ArbitrumGasData | OptimismGasData
+): Promise<{
+  gasUsedL1: BigNumber;
+  gasUsedL1OnL2: BigNumber;
+  gasCostL1USD: CurrencyAmount;
+  gasCostL1QuoteToken: CurrencyAmount;
+}> => {
+  const swapOptions: SwapOptionsUniversalRouter = {
+    type: SwapType.UNIVERSAL_ROUTER,
+    recipient: '0x0000000000000000000000000000000000000001',
+    deadlineOrPreviousBlockhash: 100,
+    slippageTolerance: new Percent(5, 10_000),
+  };
+  let l1Used = BigNumber.from(0);
```

Comment:
nit: naming is a little hard to follow here 😅 , can we do something like `mainnetGasUsed`?